src/scipp/core/data_group.py

```diff
@@ -177,7 +177,12 @@ def __setitem__(self, name: str, value: Any):
 
     def __setitem__(self, name, value):
         """Set self[key] to value."""
-        self._items[name] = value
+        if isinstance(name, str):
+            self._items[name] = value
+        else:
+            raise ValueError(
+                'Assigning to slice with implicit dimension label is not possible.'
```

Comment:
Why 'implicit dim labels'? I would expect the message to talk about the type.

Reply:
Ah, I read the example in the issue again. Maybe the message should be a little longer and be more explicit about types and slicing.

Reply:
How about simply: "Keys must be strings"?

Reply:
It shouldn't be `KeyError` because that is used for failed lookups. But otherwise, I'm fine with either message

Reply:
Okay I'll make it simply "TypeError: Keys must be strings"

Reply:
The slightly confusing thing from an api perspective is that you can get items using slicing without specifying the label:
```python
import scipp as sc
ds = sc.DataGroup(a=sc.arange('x', 4), b=sc.arange('y', 6))
ds['x', 2] # This works, returns a DataGroup
```
but if you try to set something without specifying the label
```python
ds['x', 2] = sc.scalar(6) 
```
it raises  `TypeError: Keys must be strings`.

What they should do of course is:
```python
ds['a']['x', 2] = sc.scalar(6)
```

Reply:
Agree. This is why I mentioned using a longer error message. But it might be fine the way it is

Reply:
Hmm, maybe you are right that we should mention that setting slices is not supported?